tests/collections/common/test_utils.py

```diff
@@ -31,3 +34,22 @@ def test_flatten(self):
 
         for n, test_case in enumerate(test_cases):
             assert flatten(test_case['input']) == test_case['golden'], f'Test case {n} failed!'
+
+
+class TestTextUtils:
+    @pytest.mark.unit
+    def test_get_num_lines(self):
+        """Test getting the number of lines from a text file.
+        """
+        test_cases = [0, 1, 10, 42, 97]
+
+        for num_lines in test_cases:
+
+            lines = [f'line {n}\n' for n in range(num_lines)]
+
+            with tempfile.TemporaryDirectory() as tmp_dir:
```

Comment:
Pytest can handle this natively: https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html (so just add tmp_path fixture and use it insteam of tmp_dir). It's a bit more user-friendly, because if test fails, it's not deleted right away, and you can inspect generated files.

Reply:
Did not know this, very nice given how many places we use tempfile in tests